recipe/meta.yaml

```diff
@@ -96,18 +101,21 @@ outputs:
       string: cpu_0
 
   - name: r-xgboost
-    script: install-r-xgboost.sh
+    script: install-r-xgboost.sh  # [not win]
+    script: install-win-wrapper.bat  # [win]
     build:
       rpaths:
         - lib/R/lib
     requirements:
       build:
-        - {{ compiler('c') }}
-        - {{ compiler('cxx') }}
-        - git
-        - cmake
-        - make
-        - llvm-openmp  # [osx]
+        - {{ compiler('c') }}    # [not win]
+        - {{ compiler('cxx') }}  # [not win]
+        - llvm-openmp            # [osx]
+        - m2w64-toolchain        # [win]
```

Comment:
The current error could also be fixed by using `clang` instead of MSYS2 but I would expect that to throw different errors. Still, this might be worth a try.
```suggestion
        - {{ compiler('r_clang') }}        # [win]
```